### ISSUE TITLE
Ver. 1.0.15 is flagged for 1.20.1 Fabric but is built against Java 21

### ISSUE THREAD

**Issue by Hexblood** [gitsh01/libertyvillagers#134]
1.20.1 is built on Java 17. So a Java 21 build does not work. I am not certain but I think it is 1.20.4 or 1.20.5 where Minecraft makes the change to Java 21.

I am getting this listing on Modrinth if that matters for you at all.

**Comment by Dekamir** [comment]
Have the same issue, also on Modrinth.

**Comment by Routhinator** [comment]
Full error for reference:

```
Uncaught exception in thread "main"
java.lang.RuntimeException: Error creating Mixin config libertyvillagers.mixins.json for mod libertyvillagers
	at net.fabricmc.loader.impl.launch.FabricMixinBootstrap.init(FabricMixinBootstrap.java:96)
	at net.fabricmc.loader.impl.launch.knot.Knot.init(Knot.java:151)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:68)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
Caused by: org.spongepowered.asm.launch.MixinInitialisationError: Error initialising mixin config libertyvillagers.mixins.json
	at org.spongepowered.asm.mixin.transformer.Config.create(Config.java:153)
	at org.spongepowered.asm.mixin.Mixins.createConfiguration(Mixins.java:100)
	at org.spongepowered.asm.mixin.Mixins.addConfiguration(Mixins.java:87)
	at net.fabricmc.loader.impl.launch.FabricMixinBootstrap.init(FabricMixinBootstrap.java:94)
	... 3 more
Caused by: java.lang.IllegalArgumentException: The requested compatibility level JAVA_21 could not be set. Level is not supported by the active JRE or ASM version (Java 17.0, ASM 9.6 (ASM10_EXPERIMENTAL))
	at org.spongepowered.asm.mixin.MixinEnvironment.setCompatibilityLevel(MixinEnvironment.java:1627)
	at org.spongepowered.asm.mixin.transformer.MixinConfig.initCompatibilityLevel(MixinConfig.java:557)
	at org.spongepowered.asm.mixin.transformer.MixinConfig.postInit(MixinConfig.java:502)
	at org.spongepowered.asm.mixin.transformer.MixinConfig.onLoad(MixinConfig.java:430)
	at org.spongepowered.asm.mixin.transformer.MixinConfig.create(MixinConfig.java:1293)
	at org.spongepowered.asm.mixin.transformer.Config.create(Config.java:148)
	... 6 more

```

**Comment by Routhinator** [comment]
Notable: There were 2.0.0 betas released, then this 1.0.15 release.. and it's built against Java 21. With that in mind, should folks on 1.20.1 be installing the 2.0.0 beta? Or sticking with 1.0.14 until this is fixed?

**Comment by mpustovoi** [comment]
Even the version name says it:
![image](https://github.com/gitsh01/libertyvillagers/assets/92390626/27580065-b51c-44e7-926f-60ec3d950116)

**Comment by mpustovoi** [comment]
Come on. It can't be that hard to change one field in the project version from `1.20.1` to `1.20.6`

**Comment by Routhinator** [comment]
Not sure who that was directed at, if its the mod owner.. They might be busy with end of school year or something.

If me, this version comes up with an explicit filter for 1.20.1 becausd its marked as 1.20.x compatible in the modrinth version that was created, regardless of any text in the version number.

The modrinth metadata is wrong. So while this should only come up in updates for 1.20.6 its coming up for anyone updating a modpack via a launcher consuming the modrinth API as a valid 1.20.x version.

**Comment by mpustovoi** [comment]
Unfortunately, it's still a thing.

**Comment by gitsh01** [comment]
I fixed 1.20.6 on Modrith.

So the beta vs regular thing, here's the story:
I rewrote the thing in Architectuary so that I could release on forge.
Then I went through a devastating divorce, layoffs at work (not me but folks on my team), and am currently struggling with health issues. It's been a rough two years.
I haven't had a chance to work further on the architectuary version, which is the version I'd eventually like to take forward since I can support forge.

Another user, Mechazawa, kindly did a pull request that added support for 1.20.2+, but they used the old version, not the architectuary version. Since I was/am unable to work on this, I verified their work works locally, then pushed it up. I'm not really able to maintain this mod right now, unfortunately.  I am very grateful for their pull request and am very grateful for any pull requests that help keep this going - because it's a mixin, it breaks nearly every time Minecraft does a serious update.

Please re-open if you want any version earlier than 1.20.6 fixed, otherwise I'm going to assume folks are okay with the 1.20.6 version.
